crates/sp-domains-fraud-proof/src/verification.rs

```diff
@@ -504,6 +507,21 @@ where
         return Err(InvalidBundleEquivocationError::BadBundleSignature);
     }
 
+    let operator_set_1 = (
+        header_1.header.proof_of_election.operator_id,
+        header_1.header.proof_of_election.domain_id,
+    );
+    let operator_set_2 = (
+        header_2.header.proof_of_election.operator_id,
+        header_2.header.proof_of_election.domain_id,
+    );
+
+    // Operator and the domain the proof of election targeted should be same
+    // while this check is strictly not required, we can short circuit earlier
```

Comment:
It is strictly required IMHO, plz see https://github.com/subspace/subspace/pull/2197#discussion_r1384955665